openapi-cli/src/main/java/io/ballerina/openapi/cmd/CodeGenerator.java

```diff
@@ -423,18 +456,109 @@ private void setGeneratedFileName(File[] listFiles, GenSrcFile gFile, int duplic
-        sourceFiles.add(new GenSrcFile(GenSrcFile.GenFileType.MODEL_SRC, srcPackage,  TYPE_FILE_NAME,
-                schemaContent));
+    private String modifySchemaContent(SyntaxTree schemaSyntaxTree, String clientContent, String schemaContent)
+            throws IOException, FormatterException {
+        Map<String, String> tempSourceFiles = new HashMap<>();
+        tempSourceFiles.put(CLIENT_FILE_NAME, clientContent);
+        tempSourceFiles.put(TYPE_FILE_NAME, schemaContent);
+        List<String> unusedTypeDefinitionNameList = getUnusedTypeDefinitionNameList(tempSourceFiles);
+        while (unusedTypeDefinitionNameList.size() > 0) {
+            ModulePartNode modulePartNode = schemaSyntaxTree.rootNode();
+            NodeList<ModuleMemberDeclarationNode> members = modulePartNode.members();
+            List<ModuleMemberDeclarationNode> unusedTypeDefinitionNodeList = new ArrayList<>();
+            for (ModuleMemberDeclarationNode node : members) {
+                if (node.kind().equals(SyntaxKind.TYPE_DEFINITION)) {
+                    for (ChildNodeEntry childNodeEntry : node.childEntries()) {
+                        if (childNodeEntry.name().equals("typeName")) {
+                            if (unusedTypeDefinitionNameList.contains(childNodeEntry.node().get().toString())) {
+                                unusedTypeDefinitionNodeList.add(node);
+                            }
+                        }
+                    }
+                } else if (node.kind().equals(SyntaxKind.ENUM_DECLARATION)) {
+                    for (ChildNodeEntry childNodeEntry : node.childEntries()) {
+                        if (childNodeEntry.name().equals("identifier")) {
+                            if (unusedTypeDefinitionNameList.contains(childNodeEntry.node().get().toString())) {
+                                unusedTypeDefinitionNodeList.add(node);
+                            }
+                        }
+                    }
+                }
+            }
+            NodeList<ModuleMemberDeclarationNode> modifiedMembers = members.removeAll
+                    (unusedTypeDefinitionNodeList);
+            ModulePartNode modiedModulePartNode = modulePartNode.modify(modulePartNode.imports(),
+                    modifiedMembers, modulePartNode.eofToken());
+            schemaSyntaxTree = schemaSyntaxTree.modifyWith(modiedModulePartNode);
+            schemaContent = Formatter.format(schemaSyntaxTree).toString();
+            tempSourceFiles.put(TYPE_FILE_NAME, schemaContent);
+            unusedTypeDefinitionNameList = getUnusedTypeDefinitionNameList(tempSourceFiles);
+        }
+        return schemaContent;
+    }
 
-        return sourceFiles;
+    private List<String> getUnusedTypeDefinitionNameList(Map<String, String> srcFiles) throws IOException {
+        List<String> unusedTypeDefinitionNameList = new ArrayList<>();
+        Path tmpDir = Files.createTempDirectory("openapi-tmp" + System.nanoTime());
```

Comment:
What if we create a folder as `.tmp`? Also, I couldn't find where you are deleting this temp directory.

Reply:
Updated